### PR TITLE
New feature: Support big characters on 20x4 displays

### PR DESCRIPTION
**Pull request by pgrawehr**
Adds support for displaying big characters on big LCD displays (20x4 or above). 

This is particularly useful to display numbers (e.g. Temperature, Time) on these displays.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/iot/pull/1578)

### CODE REVIEW

**Review by Ellerbach — COMMENTED** [review]
On the font side, see my comment. For the rest, looks good. Only issue is touching the interface now requires to adjust all the screens.

File: src/devices/CharacterLcd/LcdValueUnitDisplay.cs [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+                _encoding = null;
+            }
+        }
+
+        private void ReadCharacterMap()
+        {
+            _font.Clear();
+            var assembly = Assembly.GetExecutingAssembly();
+            // Get our resource file (independent of default namespace of project)
+            string resourceName = assembly.GetManifestResourceNames().Single(str => str.EndsWith("BigFontMap.txt"));
```
[the remark this answers is not in the document]
Comment (pgrawehr) [reply src/devices/CharacterLcd/LcdValueUnitDisplay.cs:121]: The font is very different from that. This actually does a kind of "brick graphics" display. It loads a fixed set of graphic primitives to the display (◢ (triangle in lower right corner) ◥ (triangle in top right corner)
◣ (triangle in lower left corner) ◤ (triangle in top left corner) ▄ (lower half block) ▀ (upper half block) █ (full block) and then the font combines these into characters of size 3x4 or 4x4.

**Comment by pgrawehr** [comment]
> 
> 
> On the font side, see my comment. For the rest, looks good. Only issue is touching the interface now requires to adjust all the screens.

Sure, I need to fix the others. The reason I had to touch the interface is testability. Methods containing parameters of Type `Span<T>` or `ReadOnlySpan<T>` cannot be mocked, because it is illegal to box these types.

**Review by Ellerbach — APPROVED** [review]
Looks good to me! Nice addition!

**Pull request merged**